File: triangles.py
def triangle(side1,side2,side3):#function to check if the sides form a triangle
    if side1==side2==side3:#if all sides are equal
        print("This is a equilateral triangle")
    elif side1==side2 and side1!= side3 or side1==side3 and side1!= side2 or side2==side3 and side2!= side1:
        print("the triangle is iscoclose")
    elif side1!=side2 and side2!= side3:#if all sides are different
        if side1*side1==side2*side2+side3*side3 or side2*side2==side1*side1+side3*side3 or side3*side3==side1*side1+side2*side2:#if the sides form a right angle
            print("This is a right triangle")
    else:#if all sides are different
        print("the triangle is invalid")

def main():
    side1=int(input("enter the length of side 1"))#inputting the sides of the triangle
    side2=int(input("enter the length of side 2"))
    side3=int(input("enter the length of side 3"))#inputting the sides of the triangle
    triangle(side1,side2,side3)
main()

File: triangles.java
import java.util.Scanner;

/**
 * triangles
 */
public class triangles {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter side one: ");
        int side1=input.nextInt();
        System.out.println("Enter side two: ");
        int side2=input.nextInt();
        System.out.println("Enter side three: ");
        int side3=input.nextInt();
        if (side1==side2 && side2==side3 && side1==side3){
            System.out.println("It is an Equilateral Triangle");
        }
        else if (side1==side2 && side2 != side3 || side2==side3 && side1 != side3 || side1==side3  && side2 != side3){
            System.out.println("It is an Isosceles Triangle");
        }
        else if (side1!=side2 && side2!=side3){
            if (side1*side1==side2*side2+side3*side3 || side2*side2==side1*side1+side3*side3 || side3*side3==side1*side1+side2*side2){
                System.out.println(" It is a Right Triangle");
        }
        } 
        else{
            System.out.println("It is a scalene Triangle");
        }
    }
}